[Partial fix] Fixed the issue for get_location_posts

otanim opened the pull request:
@aandergr, please check this PR,

it fixes the following issue:
https://github.com/instaloader/instaloader/issues/1129https://github.com/instaloader/instaloader/issues/1129

However the fix has a side effect, the API cannot retrieve the page of data via `max_id` that is being retrieved from the payload's `next_max_id`. Basically, it keeps downloading the same page over and over as if it ignores the `max_id` parameter.

Please do not merge this PR until it's not completely fixed, feel free to push new commits to this branch if you figured out the solution.

aandergr commented:
Hi! Thanks a lot for your effort so far with this PR!

I have noticed that it contains a lot of unrelated changes, such as whitespace changes in otherwise untouched parts of the code and that it does not pass the automatic status checks. We are currently managing a large number of PRs in parallel, and accordingly it would be very helpful if you could solve these superficial problems first before we take a closer look. Also, I am usually hard to convince to add new dependencies.

otanim commented:
Hey @aandergr, sure thing,
- regarding unrelated changes (whitespaces, etc), those changes happened during code refactorings, I've reverted those back,
- regarding the newly added dependency, I fully understand your concern, I've removed it (it was [pydash](https://github.com/dgilland/pydash), it was a library for simplification of actions related to objects/arrays.
- regarding not addressing the issue "TypeError: 'NoneType' object is not subscriptable", it actually does, the issue here was that the payload that is being received from Instagram, has 2 fields:
   - `graphql` which is a field of type "None",
   - and another field called `native_location_data`, which is an alternated version of the expected and outdated `graphql` field.
   
I hope after the clean-up it's much better/valid for the review, please double-check my newest changes.

otanim commented:
Hey @aandergr, previously I was wrong regarding my fix for `--geotags`, I've updated the PR with the fix.

aandergr commented on instaloader/structures.py:
```diff
@@ -553,10 +553,14 @@ def location(self) -> Optional[PostLocation]:
-                                         params={'__a': 1})['graphql']['location']
-        self._location = PostLocation(location_id, loc['name'], loc['slug'], loc['has_public_page'],
-                                      loc['lat'], loc['lng'])
+            loc_post_requested = self._context.get_json("explore/locations/{0}/".format(location_id),
+                                         params={'__a': 1})['native_location_data']['location_info']
+        loc_name = loc.get("name") or loc_post_requested.get("name")
+        loc_slug = loc.get("slug") or loc_post_requested.get("slug")
+        loc_has_public_page = loc.get("has_public_page") or loc_post_requested.get("has_public_page")
+        loc_lat = loc.get("lat") or loc_post_requested.get("lat")
+        loc_lng = loc.get("lng") or loc_post_requested.get("lng")
+        self._location = PostLocation(location_id, loc_name, loc_slug, loc_has_public_page, loc_lat, loc_lng)
```
Seems to be equivalent to (less code):

```suggestion
            loc.update(self._context.get_json("explore/locations/{0}/".format(location_id),
                                              params={'__a': 1})['native_location_data']['location_info'])
        self._location = PostLocation(location_id, loc['name'], loc['slug'], loc['has_public_page'],
                                      loc['lat'], loc['lng'])
```

aandergr reviewed (commented):
Nice, the part fixing #1109 seems to work fine, thanks! I just suggest an equivalent, but simpler, change.

Given that the fix for #1129 is still partial and not yet ready to merge, could you split up these two parts into separate PRs, so that we could already merge and release the fix for #1109? (probably it's easiest to just create a new PR for #1109 and after that remove that part from this PR)

I still did not take a closer look to the other part of the PR yet, unfortunately; but I hope I can get to that soon.

otanim commented:
@aandergr, done.

aandergr commented:
Seems the browser does an iphone request (we have implemented this request type in InstaloaderContext.get_iphone_json) to `https://i.instagram.com/api/v1/locations/{id}/sections/`, which may also be the way to request subsequent pages.

It's cool to see a working mapping from the new sections-type post response to 'legacy' Post nodes, as currently required by `Post`. This might also be handy for fixing #1080, where the response has changed to that new format. In the long term, we will probably need to change `Post` so that it can work with all types of Post representations that are currently returned by Instagram.
